scipy/sparse/linalg/dsolve/SuperLU/SRC/creadhb.c

```diff
@@ -298,11 +298,11 @@ creadhb(FILE *fp, int *nrow, int *ncol, int *nonz,
 
     register int i, numer_lines = 0, rhscrd = 0;
     int tmp, colnum, colsize, rownum, rowsize, valnum, valsize;
-    char buf[100], type[4], key[10];
+    char buf[100], type[4], key[10], *dummy;
     int sym;
 
     /* Line 1 */
-    fgets(buf, 100, fp);
+    dummy = fgets(buf, 100, fp);
```

Comment:
This silences the warning, but it'd be better to actually handle the return values I think? See https://stackoverflow.com/questions/34191613/warning-ignoring-return-value-of-fscanf-in-c

So something like:
```c
if (fgets(buf, 100, fp))
    ABORT("Nonzero return from ");
```
Or explicitly check for `NULL` as in the example of https://www.tutorialspoint.com/c_standard_library/c_function_fgets.htm

Reply:
Okay, I will add that check too. If this diff fixes all the warnings, then I would open a PR in scipy